### DISCUSSION TITLE
Model based upscale scale

### DISCUSSION THREAD

**Discussion by WASasquatch** [Comfy-Org/ComfyUI#111] [Ideas]
There is no scale for the upscaling node for model based upscaling. This means most the good models will be forcing a user to do 4x when they may not even be able to do it under their GPU (especially if using it as part of HR-Fix like I do. Occasionally I even OOM on 4090 (weirdly by just 2mb lol).

Could we get a scalar input for the upscale node for a factor? Seems these models can handle arbitrary scales too like `1.5` so it would be the best.

**Comment by comfyanonymous** [comment]
These upscale models always upscale at a fixed ratio.

You need to use the ImageScale node after if you want to downscale the image to something smaller. 
That's exactly how other UIs that let you adjust the scaling of these models do it, they downscale the image using a regular scale method after.

**Reply by WASasquatch** [reply]
I don't think that's true. For example Upscayl that uses 4x ultrasharp would output a 2x, where as if I used double (4x) it would crash cause my old GPU only had 4GB of VRAM. GFPGAN, CodeFromer, and also have scales (though CodeFormer more presets), where going above what you can do will result in error. That should be indication that the processing is at scale. 

I did just end up adding a downscale after model upscale, but some people reporting still running out of error, while able to use the same methods on WebUI.

**Reply by comfyanonymous** [reply]
If you look at the a1111 ui code you see this is the function call used to upscale with a model: https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/master/modules/upscaler.py#L63

You'll notice there's no width or height passed to it.

Instead there's this right after: https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/master/modules/upscaler.py#L71

This means it upscales with the model you pick and then scales afterwards to the exact resolution you want.

**Reply by WASasquatch** [reply]
Huh, so Comfy is just heavier on VRAM usage, alright. You are able to use scales in many other software though. I'm sure it can be done.

**Reply by comfyanonymous** [reply]
It's probably running out of memory in the VAEEncode node, if you switch it the VAEEncodeTiled the problem will likely go away.

**Reply by WASasquatch** [reply]
> It's probably running out of memory in the VAEEncode node, if you switch it the VAEEncodeTiled the problem will likely go away.

Was wondering what that was. Lol

**Reply by creeduk** [reply]
Loving VAEEncodeTiled & VAEDecodeTiled  they are allowing me to get more size with my low vram.

**Reply by WASasquatch** [reply]
> Loving VAEEncodeTiled & VAEDecodeTiled they are allowing me to get more size with my low vram.

Yeah the latest pack too @comfyanonymous did is really nice, more perfect then any I've seen in any other system. Like Even if you levels the image in photoshop to reveal the really low ends of the image, the tiling is super tight and faded nicely.
